hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -422,31 +422,25 @@ public AbfsRestOperation createPath(final String path, final boolean isFile, fin
 
     final URL url = createRequestUrl(path, abfsUriQueryBuilder.toString());
     AbfsRestOperation op = null;
-    if (abfsConfiguration.getMode() == PrefixMode.BLOB){
-      requestHeaders.add(new AbfsHttpHeader(CONTENT_LENGTH, "0"));
-      requestHeaders.add(new AbfsHttpHeader(X_MS_BLOB_TYPE, BLOCK_BLOB_TYPE));
-      op = new AbfsRestOperation(
-          AbfsRestOperationType.PutBlob,
-          this,
-          HTTP_METHOD_PUT,
-          url,
-          requestHeaders);
-    } else if (abfsConfiguration.getMode() == PrefixMode.DFS) {
-      op = new AbfsRestOperation(
-          AbfsRestOperationType.CreatePath,
-          this,
-          HTTP_METHOD_PUT,
-          url,
-          requestHeaders);
-    }
     try {
-      op.execute(tracingContext);
+      if (abfsConfiguration.getMode() == PrefixMode.BLOB) {
+        op = createPathBlob(tracingContext, path, isFile, overwrite, permission, umask, eTag);
+      } else if (abfsConfiguration.getMode() == PrefixMode.DFS) {
```

Comment:
lets have else. since we foresee BLOB and DFS only.

Reply:
Done.